tests/compilation_cache_test.py

```diff
@@ -0,0 +1,142 @@
+    if jax.lib.xla_bridge.get_backend().platform != "cpu":
+        cpu_backend = jax.lib.xla_bridge.get_backend("cpu")
+        filled_hash3 = self.get_hashed_value(cc._hash_platform, cpu_backend)
+        self.assertNotEqual(filled_hash1, filled_hash3)
+    self.assertEqual(filled_hash1, filled_hash2)
+
+  def test_hash_int(self):
+    filled_hash1 = self.get_hashed_value(cc._hash_int, 90)
+    filled_hash2 = self.get_hashed_value(cc._hash_int, 8)
+    filled_hash3 = self.get_hashed_value(cc._hash_int, 8)
+    self.assertEqual(filled_hash2, filled_hash3)
+    self.assertNotEqual(filled_hash1, filled_hash2)
+
+  def test_hash_bool(self):
+    filled_hash1 = self.get_hashed_value(cc._hash_bool, False)
+    filled_hash2 = self.get_hashed_value(cc._hash_bool, True)
+    filled_hash3 = self.get_hashed_value(cc._hash_bool, True)
+    self.assertEqual(filled_hash2, filled_hash3)
+    self.assertNotEqual(filled_hash1, filled_hash2)
+
+  def test_hash_string(self):
+    filled_hash1 = self.get_hashed_value(cc._hash_string, "foo")
+    filled_hash2 = self.get_hashed_value(cc._hash_string, "bar")
+    filled_hash3 = self.get_hashed_value(cc._hash_string, "bar")
+    self.assertEqual(filled_hash2, filled_hash3)
+    self.assertNotEqual(filled_hash1, filled_hash2)
+
+  @unittest.skipIf(jax.lib.version < (0, 1, 68), "fails with earlier jaxlibs")
+  def test_same_hash_key(self):
+    computation = jax.xla_computation(lambda x, y: x + y)(1, 1)
+    compile_options = jax.lib.xla_bridge.get_compile_options(
+                       num_replicas=1, num_partitions=1)
+    self.assertEqual(cc.get_cache_key(computation, compile_options),
+                     cc.get_cache_key(computation, compile_options))
+
+  @unittest.skipIf(jax.lib.version < (0, 1, 68), "fails with earlier jaxlibs")
+  def test_different_hash_key(self):
+    computation = jax.xla_computation(lambda x, y: x + y)(1, 1)
+    compile_options_not_filled = jax.lib.xla_bridge.get_compile_options(
+                       num_replicas=1, num_partitions=1)
+    compile_options_filled = self.filled_compile_options()
+    self.assertNotEqual(cc.get_cache_key(computation, compile_options_not_filled),
+                        cc.get_cache_key(computation, compile_options_filled))
+
+  @unittest.skipIf(jax.lib.version < (0, 1, 68), "fails with earlier jaxlibs")
+  def test_different_computations(self):
+    computation1 = jax.xla_computation(lambda x, y: x + y)(1, 1)
+    computation2 = jax.xla_computation(lambda x, y: x * y)(2, 2)
+    compile_options = jax.lib.xla_bridge.get_compile_options(
+                       num_replicas=1, num_partitions=1)
+    self.assertNotEqual(cc.get_cache_key(computation1, compile_options),
+                        cc.get_cache_key(computation2, compile_options))
+
+  def filled_compile_options(self):
+    compile_options = jax.lib.xla_bridge.get_compile_options(
+                       num_replicas=1, num_partitions=1)
```

Comment:
Can you actually manually create a CompileOptions here instead of calling `get_compile_options`? Like this: https://github.com/google/jax/blob/main/jax/lib/xla_bridge.py#L86-L88

This will make it easier to see which fields are set.